test/src/main/java/ru/spbau/karlina/test1/HashMap.java

```diff
@@ -0,0 +1,126 @@
+package ru.spbau.karlina.test1;
+
+import ru.spbau.karlina.test1.list.List;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public class HashMap<K, V> implements Map {
+    private int hashTableMax = 4;
+    private List<K, V>[] map;
+    private int sizeOfPairs;
+
+    public HashMap() {
+        map = (List<K, V>[]) new Object[hashTableMax];
+        makeInitialization(map, hashTableMax);
+    }
+
+    private void makeInitialization(List<K, V>[] array, int arraySize) {
+        for (int i = 0; i < arraySize; ++i) {
+            array[i] = new List<K, V>();
+        }
+    }
+
+    private int getHash(K key) {
+        return key.hashCode() % hashTableMax;
+    }
+
+    public int size() {
+        return sizeOfPairs;
+    }
+
+    public boolean isEmpty() {
+        if (sizeOfPairs == 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean containsKey(Object o) {
+        K key = (K)o;
+
+        int hash = getHash(key);
+
+        return map[hash].contains(key);
+    }
+
+    public boolean containsValue(Object o) {
+        return false;
+    }
+
+    public Object get(Object o) {
+        K key = (K)o;
+
+        int hash = getHash(key);
+
+        return map[hash].get(key);
+    }
+
+    public Object put(Object o, Object o2) {
+        V value = (V)o;
+        K key = (K)o2;
+
+        int hash = getHash(key);
+
+        if (map[hash].addToEnd(key, value) != null) {
+            sizeOfPairs++;
+            return value;
+        }
+
+        return null;
+    }
+
+    public Object remove(Object o) {
+        K key = (K)o;
+
+        int hash = getHash(key);
+
+        V value = map[hash].remove(key);
+        if (value != null) {
+            sizeOfPairs--;
+        }
+
+        return value;
```

Comment:
* <p>Returns the value to which this map previously associated the key,
     * or <tt>null</tt> if the map contained no mapping for the key.